host/frontend/host-orchestrator/static/js/index.js

```diff
@@ -51,51 +54,103 @@ class DeviceListApp {
-      device_to_button_map[devId] = buttonId;
-    }
-
-    for (const [devId, buttonId] of Object.entries(device_to_button_map)) {
-      let button = document.getElementById(buttonId);
-      button.addEventListener('click', evt => {
-        this.#selectDeviceCb(devId);
+      let launchBtn = document.getElementById(launchId);
+      launchBtn.addEventListener('click', evt => {
+        this.#openDeviceInNewTab(devId);
+      });
+      let showBtn = document.getElementById(showId);
+      showBtn.addEventListener('click', evt => {
+        let shown = this.#toggleDevice(devId);
+        if (shown) {
+          showBtn.textContent = "visibility";
+          showBtn.title = 'Hide device';
+        } else {
+          showBtn.textContent = "visibility_off";
+          showBtn.title = 'Show device';
+        }
       });
     }
   }
 
-  #createDeviceEntry(devId, buttonId) {
+  #createDeviceEntry(devId, showId, launchId) {
     let li = document.createElement('li');
-    li.className = 'device_entry';
-    li.title = 'Connect to ' + devId;
+    li.className = 'device-entry';
     let div = document.createElement('div');
     let span = document.createElement('span');
+    span.className = 'device-label';
     span.appendChild(document.createTextNode(devId));
-    let button = document.createElement('button');
-    button.id = buttonId;
-    button.appendChild(document.createTextNode('Connect'));
+    span.title = devId;
+    let showBtn = document.createElement('span');
+    showBtn.appendChild(document.createTextNode('visibility_off'));
+    showBtn.title = 'View device';
+    showBtn.className = 'button material-icons';
+    showBtn.id = showId;
+    showBtn.title = 'Show device';
+    let launchBtn = document.createElement('span');
+    launchBtn.appendChild(document.createTextNode('open_in_new'));
+    launchBtn.className = 'button material-icons';
+    launchBtn.id = launchId;
+    launchBtn.title = 'Launch in new tab';
     div.appendChild(span);
-    div.appendChild(button);
+    div.appendChild(showBtn);
+    div.appendChild(launchBtn);
     li.appendChild(div);
     return li;
   }
-}  // DeviceListApp
 
-window.addEventListener('load', e => {
-  let listDevicesUrl = '/devices';
-  let selectDeviceCb = deviceId => {
+  #openDeviceInNewTab(deviceId) {
     return new Promise((resolve, reject) => {
-      let client =
-          window.open(`/devices/${deviceId}/files/client.html`, deviceId);
+      let client = window.open(this.#deviceConnectUrl(deviceId), deviceId);
       client.addEventListener('load', evt => {
-        console.log('loaded');
         resolve();
       });
     });
-  };
-  let deviceListApp = new DeviceListApp({url: listDevicesUrl, selectDeviceCb});
+  }
+
+  // Returns true if the device is visible
+  #toggleDevice(devId) {
+    let id = `device-${devId}`;
+    let viewer = document.getElementById(id);
+    if (viewer) {
+      viewer.remove();
+      return false;
+    }
+    viewer = document.createElement('div');
```

Comment:
Same concern as above, can this use a `<template>`?

Reply:
done